toolkit.py

```diff
@@ -0,0 +1,196 @@
+        min = np.argmin(arr)
+        return (min, max)
+
+    def get_board_boundaries(self, image):
+        """
+        Function gets rid of low intensity pixel values in
+        color spectrum and sets them to 0. Then it calculates
+        histogram of green color in x and y axis from image.
+        Then we calculate 1-th order difference on this
+        histogram
+
+        Args:
+            image(numpy.ndarray): Loaded image
+
+        Retunrs:
+            Returns extremes from calculated histogram
+        """
+        (width, height, _) = image.shape
+        vlines = np.zeros(width)
+        hlines = np.zeros(height)
+        image[:, :, 1] = cv.inRange(image[:, :, 1],
+                                    self.lo_thresh,
+                                    self.hi_thresh)
+        vlines = cv.reduce(image[:, :, 1], 1,
+                           cv.cv.CV_REDUCE_SUM,
+                           dtype=cv.CV_32S)
+        hlines = cv.reduce(image[:, :, 1], 0,
+                           cv.cv.CV_REDUCE_SUM,
+                           dtype=cv.CV_32S)
+        vlines = map(lambda *row: list(row), *vlines)
+        y = np.diff(vlines[0])
+        x = np.diff(hlines[0])
+        return (self.__get_extremes(x), self.__get_extremes(y))
+
+    def get_table_metadata(self, image):
+        """
+        Function distorts image with median blur to
+        remove noise from image and unify colors a bit
+
+        Args:
+            image(numpy.ndarray): Loaded image
+
+        Returns:
+            Board's corner coordinates
+        """
+        process_image = cv.medianBlur(image, 5)
+        return self.get_board_boundaries(process_image)
+
+    def __get_file_name(self, path):
+        """
+        Args:
+            path(string): Path to file
+        Returns:
+            string: File name extracted from path
+        """
+        head, tail = ntpath.split(path)
+        name = tail or ntpath.basename(head)
+        return name.split('.')[0]
+
+    def write_video_metadata(self, video, meta_file):
+        """
+        Sets up a video capture and loops through all of the
+        frames. Pulls frame metadata from every frame and stores
+        it into an array. After we are done with a video we flush these
+        values to a file
+
+        Args:
+            video(string): Name of video file to load
+            meta_file(string): Name of file where metadata will be stored
+        """
+        vid = cv.VideoCapture(video)
+        if vid is None:
+            print("Couldn't load video {}".format(video), file=sys.stderr)
+            return
+        frames = []
+        while(vid.isOpened()):
+            ret, frame = vid.read()
+            if not ret:
+                break
+            (x, y) = self.get_table_metadata(frame)
+            frames.append((x, y))
+        with meta_file as csvfile:
+            for i, frame in enumerate(frames):
+                csv_writer = csv.writer(csvfile, lineterminator='\n')
```

Comment:
Is there a reason why csv_writer needs to be initialized for each row of the metadata?